jenkins_pipelines/scripts/tests/test_maintenance_json_generator.py

```diff
@@ -138,6 +142,33 @@ def test_validate_and_store_results(self, mock_print):
         # cleanup
         remove(test_output_file)
 
+    def test_read_mi_ids_from_file(self):
+        file_ids: list[str] = read_mi_ids_from_file(_TEST_MI_IDS_FILE_PATH)
+        self.assertEqual(file_ids, ['11111', '22222', '33333'])
+
+        self.assertRaises(OSError, read_mi_ids_from_file, 'file_not_found.txt')
+
+    def test_merge_mi_ids(self):
+        # no ids at all
+        sys.argv= ['maintenance_json_generator.py', '-v', '50']
+        args = parse_cli_args()
+        ids: list[str] = merge_mi_ids(args)
+        self.assertEqual(ids, [])
+        # no mi ids file
+        sys.argv= ['maintenance_json_generator.py', '-v', '50', '-i', '1234', '5678', '-e']
+        args = parse_cli_args()
+        ids: list[str] = merge_mi_ids(args)
+        self.assertEqual(ids, ['1234', '5678'])
+        # only mi ids file
+        sys.argv= ['maintenance_json_generator.py', '-v', '50', '-f', _TEST_MI_IDS_FILE_PATH, '-e']
+        args = parse_cli_args()
+        ids: list[str] = merge_mi_ids(args)
+        self.assertEqual(ids, ['11111', '22222', '33333'])
+        # ids both from flag and file
+        sys.argv= ['maintenance_json_generator.py', '-v', '50', '-f', _TEST_MI_IDS_FILE_PATH, '-i', '1234', '5678', '-e']
+        args = parse_cli_args()
+        ids: list[str] = merge_mi_ids(args)
+        self.assertEqual(ids, ['1234', '5678', '11111', '22222', '33333'])
```

Comment:
all `-i` values are smaller than this in file, I would suggest mixing them.

Reply:
If that's because you see ordered results and you want to test whether that order is kept with different IDs, I can reassure that there's no ordering involved in these steps .


All it does is put the IDS coming from  the -i flag before the ones coming from the file. The actual numerical order does not matter here.
The keys will be ordered once the final JSON output is produced.

Reply:
ok, but where we are testing that it sorts correctly?

Reply:
Anyway, we have a new feature we should test it. I am sure that parameter of this function works correctly but given that we are parsing IDs from a two parameters, or even more combinations can be involved that I am not aware of. I would check in the end if IDs from file `<1, 4, 5>` and IDs from `-i` parameter like `-i 2, 6 -i 3` (yes, double `i` parameter) it returns `1, 2, 3, 4, 5, 6`.

I would do that, but if you consider it unnecessary, you can close it.